server/pulp/server/webservices/views/repo_groups.py

```diff
@@ -0,0 +1,378 @@
+from django.core.urlresolvers import reverse
+from django.views.generic import View
+
+from pulp.common import tags
+from pulp.common.plugins import distributor_constants
+from pulp.server import exceptions as pulp_exceptions
+from pulp.server.auth import authorization
+from pulp.server.db.model.criteria import Criteria
+from pulp.server.db.model.repo_group import RepoGroup as RepoGroupModel
+from pulp.server.managers import factory as managers_factory
+from pulp.server.managers.repo.group.publish import publish as repo_group_publish
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views.util import (
+    generate_json_response, generate_json_response_with_pulp_encoder, generate_redirect_response,
+    json_body_allow_empty, json_body_required
+)
+
+
+class RepoGroupsView(View):
+    """
+    Views for all repo groups.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        Return a serialized response containing a list of repo groups.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing a serialized list of dicts, one for each repo group
+        :rtype: django.http.HttpResponse
+        """
+
+        collection = RepoGroupModel.get_collection()
+        cursor = collection.find({})
+        groups = []
+        for group in cursor:
+            group['_href'] = reverse('repo_group_resource', kwargs={'repo_group_id': group['id']})
```

Comment:
by the way, since reverse of 'repo_group_resource' is repeated 4 times you can do something like here https://github.com/pulp/pulp/commit/c30f0962563dbed33eabf3523f0ba2930658f341#diff-113ebb31dc6faa5d855700dc67532dddR162

same with 'repo_group_distributor_resource'